sdk/core/core/inc/az_span.h

```diff
@@ -63,34 +71,62 @@ az_span_sub(az_span const span, size_t const begin, size_t const end) {
  * Returns `true` if a content of the @a span is equal to a content of the @b
  * span.
  */
+AZ_NODISCARD AZ_INLINE bool az_span_is_equal_content(az_span const a, az_span const b) {
+  return a._internal.length == b._internal.length
+      && memcmp(a._internal.begin, b._internal.begin, a._internal.length) == 0;
+}
+
+/**
+ * Returns `true` if a capacity of the @a span is equal to a capacity of the @b
+ * span.
+ */
+AZ_NODISCARD AZ_INLINE bool az_span_is_equal_capacity(az_span const a, az_span const b) {
+  return a._internal.capacity == b._internal.capacity;
+}
+
+/**
+ * Returns `true` if span @b a is equal span @b b on content and capacity
+ */
 AZ_NODISCARD AZ_INLINE bool az_span_is_equal(az_span const a, az_span const b) {
-  return a.size == b.size && memcmp(a.begin, b.begin, a.size) == 0;
+  return az_span_is_equal_capacity(a, b) && az_span_is_equal_content(a, b);
 }
 
 /**
  * Returns `true` if a content of the @a span is equal to a content of the @b
  * span using case-insensetive compare.
  */
-AZ_NODISCARD bool az_span_is_equal_ignoring_case(az_span const a, az_span const b);
+AZ_NODISCARD bool az_span_is_equal_content_ignoring_case(az_span const a, az_span const b);
 
 AZ_NODISCARD AZ_INLINE bool az_span_is_overlap(az_span const a, az_span const b) {
```

Comment:
Do we need this? If so, does it need to be public?

Reply:
Most likely it should be private. But regardless, I would review the places where we use this, because now that the span also carries capacity, and overlap() only checks for length, there is a slight chance that the code that uses this may get an unexpected new meaning in cases when the spans don't overlap by contents, but their buffers do overlap. Or, may not have a problem. But I recommend to review.